src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,101 +1,108 @@
 
-  const fetchPostDetails = async () => {
-    setIsLoading(true);
-    setPostDetails(await getPostDetails(postId));
-    fetchComments();
+      setPostDetails(result);
+    }
   };
 
   useEffect(() => {
-    if (postId !== 0) {
-      fetchPostDetails();
-    }
-  }, [postId]);
+    findPost();
+  }, []);
 
-  return (
-    <div className="PostDetails" data-cy="postDetails">
-      <h2>Post details:</h2>
+  const findcomments = async () => {
+    if (postId) {
+      const result = await getComments(postId);
 
-      {isLoading
-        ? <Loader />
-        : (
-          <>
-            <section className="PostDetails__post">
-              <p>{postDetails?.title}</p>
-            </section>
+      setCommentsList(result);
+    }
+  };
 
-            <section className="PostDetails__comments">
-              {postComments.length === 0
-                ? <p>No comments</p>
-                : (
-                  <>
-                    <button
-                      type="button"
-                      className="button"
-                      onClick={() => setHidden(!isHidden)}
-                    >
-                      {`${isHidden ? 'Hide' : 'Show'} ${postComments.length} comments`}
-                    </button>
+  const deletecomment = async (id: number) => {
+    if (commentsList) {
+      await delComment(id);
+      findcomments();
+    }
+  };
 
-                    {isHidden && (
-                      <ul className="PostDetails__list">
-                        {postComments.map(comment => (
-                          <li
-                            className="PostDetails__list-item"
-                            key={comment.id}
-                          >
-                            <button
-                              type="button"
-                              className="PostDetails__remove-button button"
-                              onClick={() => {
-                                removePostComment(comment.id);
-                                setpPostComments(
-                                  postComments.filter(c => c.id !== comment.id
-                                  ));
-                              }}
-                            >
-                              X
-                            </button>
-                            <p>{comment.body}</p>
-                          </li>
-                        ))}
-                      </ul>
-                    )}
-                  </>
-                )}
-            </section>
+  useEffect(() => {
+    findcomments();
+  }, [postId]);
 
-            <section>
-              <div className="PostDetails__form-wrapper">
-                <NewCommentForm
-                  postId={postId}
-                  fetchComments={fetchComments}
-                />
-              </div>
-            </section>
-          </>
-        )}
+  return (
+    <div className="PostDetails">
+      <h2>Post details:</h2>
+      {postId !== 0 && (
```

Comment:
You can use a more explicit condition to check if postId is truthy (i.e., not undefined, null, 0, false, or an empty string). 

Just use:
`{postId && (` instead of `{postId !== 0 && (`